### PR TITLE
Switch to poetry for dependency management; update deps

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review

## Description of Changes

- Switches to Poetry 1.4.0 (just released) for dependency management consistent with https://github.com/freedomofpress/securedrop-engineering/blob/main/proposals/approved/use-poetry.md
- Updates all dependencies along with it

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [x] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[tool.poetry]
+name = "securedrop-docs"
+version = "2.5.2"
+description = "SecureDrop documentation for journalists, sources and administrators"
+authors = ["SecureDrop team <securedrop@freedom.press>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+sphinx = "^6.1.3"
```
Comment (legoktm) [inline pyproject.toml:10]: In redmine_docs, you had:
```
# The RTD theme does not yet support Sphinx >=6.0.0.
# See https://github.com/readthedocs/sphinx_rtd_theme/issues/1323
sphinx = "<6.0.0"
```

Is that not necessary in this repo?

File: deploy/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+FROM python:3.9-slim-bullseye@sha256:8b0502fe3a8ae9d26567f66e217223595ff6abe901cba64fd8b1b58176311b27 AS sphinx
 
 ARG GIT_BRANCH=main
 RUN apt-get -q update && apt-get -qy upgrade && apt-get -qy install git make latexmk texlive-latex-extra
 COPY ./ .
-RUN pip install -r requirements/requirements.txt
-RUN deploy/build $GIT_BRANCH
+RUN pip install poetry==1.4.0
+RUN poetry install
+# TODO: Once the latest stable tag uses poetry, we can drop the `poetry run` prefix
```
Comment (legoktm) [inline deploy/Dockerfile:9]: I don't understand this comment, why are the `poetry run` in the Makefile not sufficient?

Comment (eloquence) [reply deploy/Dockerfile:9]: This repo uses a more complex versioning strategy than `redmine_docs`. Here, a git checkout of the most recent version is performed to render the stable version. That version doesn't have the required `Makefile` changes yet which wrap the `sphinx-build` invocations, so it'll error out during the build process

File: pyproject.toml (hunk 1 above)
Comment (eloquence) [reply pyproject.toml:10]: I didn't hit any compatibility issues with the most recent versions. I'll see if those have been resolved and we can update `redmine_docs` as well.

File: deploy/Dockerfile (hunk 2 above)
Comment (legoktm) [reply deploy/Dockerfile:9]: OK, I re-read `deploy/build` and now I remember how it works. Ack!

**Review by legoktm — APPROVED** [review]
Tested with poetry 1.4.0, everything looks good to me!

**Pull request merged**